### PR TITLE
[HW] Added methods to operate on in and inout ports

### PR DESCRIPTION
**Pull request by nandor**

### CODE REVIEW

File: include/circt/Dialect/HW/HWStructure.td [hunk 1]
```diff
@@ -87,6 +87,9 @@ def HWModuleOp : HWModuleOpBase<"module",
       return getAllModulePortInfos(*this);
     }
 
+    /// Returns the number of in or input ports.
+    size_t getNumInOrInoutPorts() { return getArgumentTypes().size(); }
```
Comment (fabianschuiki) [inline include/circt/Dialect/HW/HWStructure.td:91]:
```suggestion
    /// Returns the number of in or inout ports.
    size_t getNumInOrInoutPorts() { return getArgumentTypes().size(); }
```

**Review by fabianschuiki — APPROVED** [review]
This looks useful to me, and it seems to complement nicely with the output modification helpers. :+1:

File: lib/Dialect/HW/HWOps.cpp [hunk 2]
```diff
@@ -1133,6 +1136,25 @@ LogicalResult HWModuleOp::verify() { return verifyModuleCommon(*this); }
 
 LogicalResult HWModuleExternOp::verify() { return verifyModuleCommon(*this); }
 
+std::pair<StringAttr, BlockArgument>
+HWModuleOp::insertInput(unsigned index, StringAttr name, Type ty) {
+  // Find a unique name for the wire.
```
Comment (mikeurbach) [inline lib/Dialect/HW/HWOps.cpp:1141]: nit: s/wire/port/ ?

**Review by mikeurbach — COMMENTED** [review]
This is great, and I'd use it in an upcoming pass. Just some comments about the comments.

**Pull request merged**